aptos-move/e2e-tests/src/data_store.rs

```diff
@@ -132,3 +134,44 @@ impl TStateView for FakeDataStore {
         InMemoryStateView::new(self.state_data.clone())
     }
 }
+
+// These traits allow `FakeDataStore` act as an executor view as well.
```

Comment:
Does it make sense to provide default implementation of these traits for anything that implements TStateView? or is it too error-prone?

Reply:
I was thinking about that, but then we will have implicit conversions from state view to executor view and it seems it can be abused, and so that going through adapter is the only way. We can revisit if this is too limiting?

Reply:
Then let's consider just using AsExecutorView trait for FakeDataStore, if possible? (either call when needed, since it's not too expensive, or create an executor_view from it and use that everywhere, as long as the state view part isn't directly needed).
If it starts looking uglier than having this code here, then ignore :)